index 44bc96e..629ba02 100644
--- a/scripts/nix-profile.sh.in
+++ b/scripts/nix-profile.sh.in
@@ -1,3 +1,6 @@
-if [ -n "$HOME" ] && [ -n "$USER" ]; then
+if [ -n "$HOME" ] && \
+   [ -n "$USER" ] && \
+   [ "$(whoami)" != root ] && \
+   [ "$(whoami)" != portage ] ; then

     # Set up the per-user profile.
diff --git a/./scripts/nix-profile.sh.in.orig b/./scripts/nix-profile.sh.in
index 44bc96e..e2d742e 100644
--- a/scripts/nix-profile.sh.in
+++ b/scripts/nix-profile.sh.in
@@ -59,9 +59,9 @@ if [ -n "$HOME" ] && [ -n "$USER" ]; then
     # pick up `.nix-profile/share/man` because is it close to `.nix-profile/bin`
     # which is in the $PATH. For more info, run `manpath -d`.
     if [ -n "${MANPATH-}" ]; then
-        export MANPATH="$NIX_LINK/share/man:$MANPATH"
+        export MANPATH="${MANPATH}:${NIX_LINK}/share/man"
     fi

-    export PATH="$NIX_LINK/bin:$PATH"
+    export PATH="${PATH}:${NIX_LINK}/bin"
     unset NIX_LINK NIX_LINK_NEW
 fi
